ios/SingularBridge.m

```diff
@@ -0,0 +1,181 @@
+    if(customUserId){
+        [Singular setCustomUserId:customUserId];
+    }
+
+    [Singular startSession:apikey withKey:secret];
+}
+
+RCT_EXPORT_METHOD(initWithSingularLinks:(NSString*)apikey
+                  secret:(NSString*)secret
+                  customUserId:(NSString*)customUserId){
+    if(customUserId){
+        [Singular setCustomUserId:customUserId];
+    }
+
+    isSingularLinksEnabled = YES;
+    eventEmitter = self;
+
+    [Singular startSession:apikey
+                   withKey:secret
+          andLaunchOptions:launchOptions
+   withSingularLinkHandler:^(SingularLinkParams * params){
+        [SingularBridge handleSingularLinks:params];
+    }];
+}
+
+RCT_EXPORT_METHOD(setCustomUserId:(NSString*)customUserId){
+    [Singular setCustomUserId:customUserId];
+}
+
+RCT_EXPORT_METHOD(unsetCustomUserId){
+    [Singular unsetCustomUserId];
+}
+
+RCT_EXPORT_METHOD(event:(NSString*)eventName){
+    [Singular event:eventName];
+}
+
+RCT_EXPORT_METHOD(eventWithArgs:(NSString*)eventName args:(NSString*)args){
+    if(!args){
+        [Singular event:eventName];
+    }
+
+    [Singular event:eventName withArgs:[SingularBridge jsonToDictionary:args]];
+}
+
+RCT_EXPORT_METHOD(revenue:(NSString*)currency amount:(double)amount){
+    [Singular revenue:currency amount:amount];
+}
+
+RCT_EXPORT_METHOD(revenueWithArgs:(NSString*)currency amount:(double)amount args:(NSString*)args){
+    if(!args){
+        [Singular revenue:currency amount:amount];
+    }
+
+    [Singular revenue:currency amount:amount withAttributes:[SingularBridge jsonToDictionary:args]];
+}
+
+RCT_EXPORT_METHOD(customRevenue:(NSString*)eventName currency:(NSString*)currency amount:(double)amount){
+    [Singular customRevenue:eventName currency:currency amount:amount];
+}
+
+RCT_EXPORT_METHOD(customRevenueWithArgs:(NSString*)eventName currency:(NSString*)currency amount:(double)amount args:(NSString*)args){
+    if(!args){
+        [Singular customRevenue:eventName currency:currency amount:amount];
+    }
+
+    [Singular customRevenue:eventName currency:currency amount:amount withAttributes:[SingularBridge jsonToDictionary:args]];
+}
+
+RCT_EXPORT_METHOD(setUninstallToken:(NSString*)token){
+    [Singular registerDeviceTokenForUninstall:[token dataUsingEncoding:NSUTF8StringEncoding]];
+}
+
+RCT_EXPORT_METHOD(trackingOptIn){
+    [Singular trackingOptIn];
+}
+
+RCT_EXPORT_METHOD(trackingUnder13){
+    [Singular trackingUnder13];
+}
+
+RCT_EXPORT_METHOD(stopAllTracking){
+    [Singular stopAllTracking];
+}
+
+RCT_EXPORT_METHOD(resumeAllTracking){
+    [Singular resumeAllTracking];
+}
+
+RCT_REMAP_METHOD(isAllTrackingStopped, resolver: (RCTPromiseResolveBlock)resolve
+                 rejecter:(RCTPromiseRejectBlock)reject){
+    resolve([Singular isAllTrackingStopped] ? @YES : @NO);
+}
+
+RCT_EXPORT_METHOD(setReactSDKVersion:(NSString*)wrapper version:(NSString*)version){
```

Comment:
should we put this under "private methods" too?

Reply:
I don't think so... i think it's best if all the react native methods should be grouped together

Reply:
I agree with you, just thought `setReactSDKVersion` is not something we technically want customers to call right? maybe it should have a separate section? (just want to give a better indication it's not for customer use...)

Reply:
We are not exposing this method in the JS interface so it should be grouped with the other bridged methods